test/test_parser.cpp

```diff
@@ -6405,4 +6405,69 @@ TEST(Parser_issue6831)
     CHECK_EQUAL(q.count(), 1);
 }
 
+TEST(Parser_issue7393)
+{
+    Group g;
+    auto table = g.add_table_with_primary_key("Table", type_String, "id");
+    auto col_value = table->add_column(type_Mixed, "value");
+    auto obj = table->create_object_with_primary_key("EMBEDDED");
+    obj.set_json(col_value, R"([
+        [4, 5, 6],
+        {
+            "key1": 7,
+            "key2": 8,
+            "key3": [9]
+        }
+    ])");
+    auto q = table->query("value[*][*] == {4, 5, 6}");
+    CHECK_EQUAL(q.count(), 1);
+    q = table->query("value[0][*] == {4, 5, 6}");
+    CHECK_EQUAL(q.count(), 1);
+    q = table->query("value[1][*] == {4, 5, 6}");
+    CHECK_EQUAL(q.count(), 0);
+}
+
+TEST(Parser_Wildcard)
+{
+    Group g;
+    auto table = g.add_table_with_primary_key("table", type_String, "id");
+    auto origin = g.add_table_with_primary_key("origin", type_String, "id");
+    auto col_any = table->add_column(type_Mixed, "value");
+    auto col_dict = table->add_column_dictionary(type_Mixed, "dict");
+    auto col_list = table->add_column_list(type_Mixed, "list");
+    auto col_link = origin->add_column_list(*table, "link");
+
+    auto obj1 = table->create_object_with_primary_key("obj1");
+    obj1.set_json(col_any, R"([[3, 2, 1], [4, 5, 6]])");
+    obj1.set_json(col_dict, R"({"key1": 1, "key2": 2, "key3": [4, 5, 6]})");
+    obj1.set_json(col_list, R"([4, 5, 6])");
+    auto obj2 = table->create_object_with_primary_key("obj2");
+    obj2.set_json(col_any, R"([[9, 8, 7], [1, 2, 3]])");
+    obj2.set_json(col_dict, R"({"key1": 3, "key2": 4, "key3": [3, 2, 1]})");
+    obj2.set_json(col_list, R"([1, 2, 3])");
+
+    auto obj = origin->create_object_with_primary_key("top");
+    auto ll = obj.get_linklist(col_link);
+    ll.add(obj1.get_key());
+    ll.add(obj2.get_key());
+
+    auto q = table->query("value[*][*] = {1, 2, 3, 4, 5, 6}");
+    CHECK_EQUAL(q.count(), 0);
+    q = table->query("value[*][*] = {3, 2, 1}");
+    CHECK_EQUAL(q.count(), 1);
+    q = table->query("{3, 2, 1} = value[*][*]");
+    CHECK_EQUAL(q.count(), 1);
+    q = table->query("value[*][*] = dict[*][*]");
+    CHECK_EQUAL(q.count(), 1);
+    q = origin->query("link.dict.key1 = {1, 3}");
+    CHECK_EQUAL(q.count(), 0);
+    q = origin->query("link.dict.key1 = {1}");
+    CHECK_EQUAL(q.count(), 1);
+
+    q = origin->query("link[0].list = {4, 5, 6}");
+    CHECK_EQUAL(q.count(), 1);
+    q = origin->query("link.list = {4, 5, 6}");
+    // CHECK_EQUAL(q.count(), 1); // Fails
```

Comment:
The tricky thing is to define what the correct behaviour should be. The current behaviour is to try to do an element by element comparison on {4, 5, 6, 1, 2, 3} vs {4, 5, 6}. This is clearly broken and I think we should change it to use the same design that you have introduced in this PR. We could debate on if this is a breaking change or a fix, but I'd vote to call it a fix.